src/metabase/server/middleware/auth.clj

```diff
@@ -31,6 +32,14 @@
   "When set, this API key is required for all API requests."
   :visibility :internal)
 
+(def mb-api-key-doc-url
+  "Url for documentation on how to set MB_API_KEY."
+  "https://www.metabase.com/docs/latest/configuring-metabase/environment-variables#mb_api_key")
+
+(def api-key-not-set-message
+  "Message sent if an endpoint protected by this middleware is hit but MB_API_KEY is not set."
+  (format "MB_API_KEY is not set. See %s for details" mb-api-key-doc-url))
```

Comment:
I think this needs to be translated. It should use `tru` instead of `format` and be a `defn-`, or in the body of `enforce-api-key`.

See other error messages as examples:
https://github.com/metabase/metabase/blob/13e9e8ef943dd1a5ff7d6000899c16b0edefb96c/src/metabase/api/common.clj#L163

Reply:
shouldn't it be `deferred-tru`?

Reply:
You could use `deferred-tru`, yes. The examples above have both options. You could use a `def` with `deferred-tru` and call `str` on it later, instead of `defn-` with `tru`. I don't think there's any difference between the two. I wouldn't make a strong case for choosing one over the other though, other than defaulting to use `tru` if possible.

Reply:
Good call out on i18n. I ended up using a `deferred-trs`. Deferred because it's in a `def` and `trs` because we don't have a logged in user in the routes used by this middleware. It's an alternative auth from a logged in user (it's an api key) so we would fall back to the server locale in either case.